### DISCUSSION TITLE
Successful Cleanup does not delete old backups

### DISCUSSION THREAD

**Discussion by tatenalan** [spatie/laravel-backup#1601] [Q&A]
Hello community! I am using "spatie/laravel-backup": "^6.16" in a Laravel 8 project. My cleanup setup is by default. I have the following schedules commands on 

My  Kernel.php:

> protected function schedule(Schedule $schedule)
>     {
>         $schedule->command('backup:clean')->daily()->at('22:30');
>         $schedule->command('backup:run')->daily()->at('22:35');
> }

I have a cron configured on my server that runs  schedule:run every minute. Everything works great, The backups are successful, they are saved in Storage/App/Laravel folder and I get the successful emails (from both scgeducommands).

The email I get everyday of backup:clean command

> Hello!
> The clean up of the Laravel backups on the disk named local was successful.
> Application name: Laravel
> Backup name: Laravel
> Disk: local
> Newest backup size: 16.9 MB
> Number of backups: 17
> Total storage used: 276.86 MB
> Newest backup date: 2022/10/27 22:35:03
> Oldest backup date: 2022/10/11 22:35:03
>  Regards,
>  Laravel 

The email I get everyday of backup:run command 

>  Hello!
> Great news, a new backup of Laravel was successfully created on the disk named local.
> Application name: Laravel
> Backup name: Laravel
> Disk: local
> Newest backup size: 16.97 MB
> Number of backups: 18
> Total storage used: 293.83 MB
> Newest backup date: 2022/10/28 22:35:02
> Oldest backup date: 2022/10/11 22:35:03
> Regards,
> Laravel

The problem is that old backups do not get deleted. I stacked 18 already and it should stack only 16 according to Backup.php cleanup config by default. What should I do to only keep the backups from the last 16 days? I do not get any log error or email error.

Thank you in advance!

**Comment by erikn69** [comment]
> I  am using "spatie/laravel-backup": "^6.16"

Version 6.x has no support, upgrade to **Laravel Backup 8.x**

**Reply by tatenalan** [reply]
Okay I will! 
I think I solved it anyway. I changed the backup.php file so the default_strategy array looks like this and it is working:

> 'default_strategy' => [
            'keep_all_backups_for_days' => 16,        
            'keep_daily_backups_for_days' => 0,
            'keep_weekly_backups_for_weeks' => 0,
            'keep_monthly_backups_for_months' => 0,
            'keep_yearly_backups_for_years' => 0,
            'delete_oldest_backups_when_using_more_megabytes_than' => 5000,
        ],

Thank you!

**Reply by tatenalan** [reply]
I am using PHP 7.4 and Laravel Backup 8.x needs a higher PHP version
